### PR TITLE
Add Java docs links

### PR DESCRIPTION
**Pull request by aarohib**
## What does this PR do?
Adds links to the Java docs in App Dev guide

### CODE REVIEW

File: docs/workflows.md [hunk 1]
```diff
@@ -533,14 +533,16 @@ Each Workflow Execution within the series is considered a Run.
...
 
-The Temporal Server spawns the next Run only after the current Run has Completed, Failed, or Timed Out.
-This means that, if a Retry Policy has also been provided, and a Run Fails or Times Out, the Run will first be retried per the Retry Policy until the Run Completes or the Retry Policy has been exhausted.
-If the next Run, per the Cron Schedule, is due to spawn while the current Run is still Open (including retries), the Server skips the next scheduled Run.
-A [Workflow Run Timeout](#workflow-run-timeout) is used to limit the maximum amount of time of individual Runs.
-Again, if the Workflow Run Timeout is reached and there is an associated Retry Policy, the Workflow is retried before the next Cron Scheduled spawn occurs.
+The Temporal Server spawns the next Run only after the current Run has Completed, Failed, or has reached the Workflow Run Timeout.
+This means that, if a Retry Policy has also been provided, and a Run Fails or reaches the Workflow Run Timeout, the Run will first be retried per the Retry Policy until the Run Completes or the Retry Policy has been exhausted.
```
Comment (djmagee) [inline docs/workflows.md:540]:
```suggestion
The Temporal Server spawns the next Run only after the current Run Completes, Fails, or reaches the Workflow Run Timeout.
This means that, if a Retry Policy has also been provided, and a Run Fails or reaches the Workflow Run Timeout, the Run is first retried per the Retry Policy until the Run Completes or the Retry Policy has been exhausted.
```

File: docs/workflows.md [hunk 2]
```diff
@@ -533,14 +533,16 @@ Each Workflow Execution within the series is considered a Run.
 - Each Run inherits the same Workflow Options as the initial Run.
 
 The Temporal Server spawns the first Workflow Execution in the chain of Runs immediately.
-However, it calculates and applies a backoff so that the first Workflow Task of the Workflow Execution does not get placed into a Task Queue until the scheduled time.
-After each Run Completes, Fails, or Times Out, the same thing happens: the next run will be created immediately, but with a backoff.
+However, it calculates and applies a backoff (`firstWorkflowTaskBackoff`) so that the first Workflow Task of the Workflow Execution does not get placed into a Task Queue until the scheduled time.
+After each Run Completes, Fails, or reaches the [Workflow Run Timeout](#workflow-run-timeout), the same thing happens: the next run will be created immediately with a new `firstWorkflowTaskBackoff` that is calculated based on the current Server time and the defined Cron Schedule.
```
Comment (djmagee) [inline docs/workflows.md:537]:
```suggestion
After each Run Completes, Fails, or reaches the [Workflow Run Timeout](#workflow-run-timeout), the same thing happens: the next run is created immediately with a new `firstWorkflowTaskBackoff` that is calculated based on the current Server time and the defined Cron Schedule.
```

**Review by djmagee — APPROVED** [review]

File: docs/application-development-guide.md [hunk 3]
```diff
@@ -1771,7 +2108,123 @@ workflowRun, err := c.ExecuteWorkflow(context.Background(), workflowOptions, "Yo
...
+`DynamicWorkflow` can be used to invoke different Workflow Types.
+To check what type is running when your Dynamic Workflow `execute` method runs, use `getWorkflowType()` in the implementation code.
+
+```java
+String type = Workflow.getInfo().getWorkflowType();
+```
+
+See [Workflow Execution Result](#get-workflow-results) for details on how to get the results of the Workflow Execution.
+
+#### Using `ExternalWorkflowStub`
```
Comment (flossypurse) [inline docs/application-development-guide.md:2210]: Can't have these headers in the source content.

Comment (aarohib) [reply docs/application-development-guide.md:2210]: updated across.

**Comment by aarohib** [comment]
All updates in.

**Pull request closed without merging**